lumicks/pylake/nb_widgets/detail/undo_redo.py

```diff
@@ -0,0 +1,24 @@
+class UndoableState:
+    """Keeps a state with an undo and redo list."""
+
+    def __init__(self, initial_state):
+        self.state = initial_state
```

Comment:
I believe `state` should not be public to keep the class invariant. I suggest going with `._state` and with maybe a property and setter?

Reply:
Right. As we discussed on meet, I'll pull it in.